src/actions/hooks.js

```diff
@@ -1,89 +1,185 @@
- * @returns status
+ * Wrapper for hooks to use certain GET endpoints
+ * For the function used by the components:
+ * @param {Number} id Single ID to fetch, array of IDs to fetch data for, or undefined/null/falsey to fetch all (see second parameter)
+ * @param {Boolean} noFetchOnUndef If true, undefined/null/falsey does NOT fetch all (nothing is fetched). Useful with an ID that may be undefined until some other data comes in, in order to avoid a fetch all being called (Optimization).
+ * @returns {Number} status of the request (null if not started/canceled)
  */
-export const useStudent = (studentID) => {
+export const wrapUseEndpoint = (endpoint, successType) => (id, noFetchOnUndef) => {
     const token = useSelector(({auth}) => auth.token);
-    const [status, setStatus] = useState(REQUEST_STARTED);
+    const [status, setStatus] = useState(null);
     const dispatch = useDispatch();
     useEffect(() => {
         let aborted = false;
-        if (!Array.isArray(studentID)) {
+        if (!id) {
+            if (!noFetchOnUndef) {
+                (async () => {
+                    try {
+                        setStatus(REQUEST_STARTED);
+                        const response = await instance.get(
+                            endpoint,
+                            {
+                                "headers": {
+                                    "Authorization": `Token ${token}`,
+                                },
+                            }
+                        );
+                        if (!aborted) {
+                            dispatch({
+                                "type": successType,
+                                "payload": {
+                                    "id": REQUEST_ALL,
+                                    response,
+                                },
+                            });
+                            setStatus(response.status);
+                        }
+                    } catch (err) {
+                        if (!aborted) {
+                            setStatus((err && err.response && err.response.status) || MISC_FAIL);
+                        }
+                    }
+                })();
+            }
+        } else if (!Array.isArray(id)) {
             (async () => {
-                const requestURL = studentID ? `/account/student/${studentID}/` : "/account/student/";
-
                 try {
-                    const response = await instance.get(requestURL, {
-                        "headers": {
-                            "Authorization": `Token ${token}`,
-                        },
-                    });
+                    setStatus(REQUEST_STARTED);
+                    const response = await instance.get(
+                        `${endpoint}${id}/`,
+                        {
+                            "headers": {
+                                "Authorization": `Token ${token}`,
```

Comment:
allow users to provide their own heading/configuration if they choose to put that as a param in the function.